balance/balance_contract.go

```diff
@@ -63,31 +59,64 @@ func _deploy(data interface{}, isUpdate bool) {
 	ctx := storage.GetContext()
 	if isUpdate {
 		args := data.([]interface{})
-		common.CheckVersion(args[len(args)-1].(int))
+		version := args[len(args)-1].(int)
+
+		common.CheckVersion(version)
+
+		if args[0].(bool) {
+			panic("update to non-notary mode is not supported anymore")
+		}
+
+		// switch to notary mode if version of the current contract deployment is
+		// earlier than v0.17.0 (initial version when non-notary mode was taken out of
+		// use)
+		// TODO: avoid number magic, add function for version comparison to common package
+		if version < 17_000 {
+			switchToNotary(ctx)
+		}
+
 		return
 	}
 
 	args := data.(struct {
 		notaryDisabled bool
-		addrNetmap     interop.Hash160
-		addrContainer  interop.Hash160
+		_, _           interop.Hash160 // Netmap and Container contract from non-notary legacy
 	})
 
-	if len(args.addrNetmap) != interop.Hash160Len || len(args.addrContainer) != interop.Hash160Len {
-		panic("incorrect length of contract script hash")
+	if args.notaryDisabled {
```

Comment:
If we no longer need any parameters for a contract, let's drop this code completely (and don't touch `data` at all). This applied to any contract that needs no parameters (others have to keep the same structure for now, so checking `notaryDisabled` there makes sense).